### ISSUE TITLE
No error output on windows node

### ISSUE THREAD

**Issue by z0c** [chef/chef#3343]
I keep getting the following message whenever a error is raised on my cookbook: 
```
==> default: [2015-05-07T13:01:04+01:00] INFO: Running queued delayed notifications before re-raising exception
==> default:
==> default: [2015-05-07T13:01:04+01:00] ERROR: Running exception handlers
==> default: [2015-05-07T13:01:04+01:00] ERROR: Exception handlers complete
==> default: [2015-05-07T13:01:04+01:00] FATAL: Stacktrace dumped to C:/var/chef/cache/chef-stacktrace.out
==> default: [2015-05-07T13:01:04+01:00] FATAL: ArgumentError: too short control escape
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
```

Seems that Chef solo is not handling the backslashes on physical paths on a windows node

Chef client version is 12.3.0 and the node is running Windows 2012 R2.

Here is the full stack trace:
```
Generated at 2015-05-07 13:01:04 +0100
ArgumentError: too short control escape
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_inspectors/resource_failure_inspector.rb:66:in `recipe_snippet'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_inspectors/resource_failure_inspector.rb:43:in `add_explanation'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_mapper.rb:72:in `resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/base.rb:139:in `resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `block (2 levels) in def_forwarding_method'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `block in def_forwarding_method'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:574:in `rescue in run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:583:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:339:in `block in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `catch'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:353:in `converge_and_save'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:457:in `run'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:253:in `run_with_graceful_exit_option'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:230:in `block in run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:213:in `run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:280:in `block in interval_run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `loop'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `interval_run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:247:in `run_application'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
C:/opscode/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
C:/opscode/chef/bin/chef-solo:63:in `load'
C:/opscode/chef/bin/chef-solo:63:in `<main>'
```

**Comment by jdmundrawala** [comment]
Can you paste your recipe? It looks like you didn't properly escape your `\` from the error message

**Comment by z0c** [comment]
The recipe is very simple:
```
windows_feature 'NetFx3' do
	action :install
end
```

DISM fails as the source is not available on windows 2012 R2, but I never get any output back

Im getting the same output from the following when the service fails to start
```
service 'service-name' do
  action [:enable, :start]
end
```

**Comment by jdmundrawala** [comment]
Do any resources in your recipe ever successfully execute?

**Comment by z0c** [comment]
yes, I changed it to the following:
```
windows_feature 'NET-Framework-Core' do
  provider Chef::Provider::WindowsFeaturePowershell
  action :install
end
```
which works on 2012 R2 and the recipe executes with no issues. It's only when an error is raised during the execution of the recipe that I get this problem

**Comment by lamont-granquist** [comment]
fixed by #3357?
